src/core/pipeline/pipeline.js

```diff
@@ -0,0 +1,35 @@
+export default class Pipeline {
+
+	constructor(stages) {
+		this.stages = stages || [];
+	}
+
+	fetch(ctx, data = []) {
```

Comment:
```javascript
pipe
    .run()
    .then((data) => ...)
```

In context of grid pipeline performs a kind of `read` operation. As for me `run` could be good for a generic middleware that doesn't return anything.

We can implement generic async middleware with `run` and then invoke `pipe.read` which internally will use `middleware.run`